scripts/diff_studies.py

```diff
@@ -0,0 +1,187 @@
+            # Therefore, sort them so that when compared, no errors are triggered
+            for i in aggr_head:
+                data[i] = ';'.join(sorted(data[i].split(';')))
+            # two loops, for up until primary key, then after.
+            for i in range(len(data)):
+                if i == primary: continue
+                data_dict[data[primary]][header[i]] = data[i]
+    attr_set = set(header)
+    # no need for primary key to be reported as an attribute
+    attr_set.remove(key)
+    return data_dict, attr_set
+
+
+def data_clinical_from_study(cbio_conn, study_id, data_type, aggr_list):
+    """
+    Get all the column-value pairs for each data_type(SAMPLE or PATIENT) for a specific study
+    Convert result to dict
+    Also return a set of attribute keys
+    """
+    # object is a big ass array of struct, one entry per attribute, per patient/sample - so like if a table were just concatenated into a single vector 
+    data_clinical = cbio_conn.Clinical_Data.getAllClinicalDataInStudyUsingGET(studyId=study_id, projection='DETAILED', clinicalDataType=data_type).result()
+    data_dict = {}
+    # Use sampleId or patientID, clinicalAttributeId (column name) and value
+    attr_dict = {"SAMPLE": "sampleId", "PATIENT": "patientId" }
+    status = ["OS_STATUS"]
+    for entry in data_clinical:
+        clinical_id = getattr(entry, attr_dict[data_type])
+        if clinical_id not in data_dict:
+            # every entry per sample as sampleId and patientId, patient just patientId. Add keys to match
+            data_dict[clinical_id] = {"PATIENT_ID": entry.patientId}
+        value = entry.value
+        attr_id = entry.clinicalAttributeId
+        # For ease of comparison, aggregate attributes concatenated by a separator may not be in the same order, but order does not matter
+        # Therefore, sort them so that when compared, no errors are triggered
+        if attr_id in aggr_list:
+            value = ';'.join(sorted(value.split(';')))
+        # "standardize" status field so that 0:LIVING = LIVING and 1:DECEASED = DECEASED
+        if attr_id in status:
+            value = value[2:]
+        data_dict[clinical_id][attr_id] = value
+    return data_dict
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Compare local clinical data to server"
+    )
+    parser.add_argument(
+        "-u", "--url", action="store", dest="url", help="url to search against", default="https://pedcbioportal.kidsfirstdrc.org/api/v2/api-docs"
+    )
+    parser.add_argument(
+        "-s", "--study", action="store", dest="study", help="Cancer study ID to compare on server"
+    )
+    parser.add_argument(
+        "-t", "--token", action="store", dest="token", help="Token file obtained from Web API"
+    )
+    parser.add_argument(
+        "-d", "--datasheet-dir", action="store", dest="data_dir", help="Directory containing data_clinical_*.txt"
+    )
```

Comment:
So why not just have this be a file input? That would save you the awkward hardcoding of the filename later in the script.